src/util/saveArticles.ts

```diff
@@ -118,26 +118,62 @@ async function downloadBulk(listOfArguments: any[], downloader: Downloader): Pro
                     return resp;
                 });
             },
-            {concurrency: CONCURRENCY_LIMIT}
+            { concurrency: CONCURRENCY_LIMIT }
         );
     } catch (err) {
         logger.log(`Not able download in bulk due to ${err}`);
     }
 }
 
+async function getAllArticlesToKeep(downloader: Downloader, mw: MediaWiki, dump: Dump) {
+    if (!dump.customProcessor) {
+        return;
+    }
+    await articleDetailXId.iterateItems(
+        downloader.speed,
+        async (articleKeyValuePairs) => {
+            for (const [articleId] of Object.entries(articleKeyValuePairs)) {
+                try {
+                    const rets = await downloader.getArticle(articleId, dump);
+
+                    for (const { articleId, html: articleHtml } of rets) {
+                        if (!articleHtml) {
+                            continue;
+                        }
+
+                        const { articleDoc: _articleDoc } = await processArticleHtml(articleHtml, downloader, mw, dump, articleId);
+                        const articleDoc = _articleDoc;
+
+                        if (dump.customProcessor?.shouldKeepArticle) {
+                            const shouldContinue = await dump.customProcessor.shouldKeepArticle(articleId, articleDoc);
+                            if (!shouldContinue) {
+                                await articleDetailXId.delete(articleId);
+                            }
```

Comment:
There is probably a more elegant manner to code this (in one line). Considering that we don't expect anything from the `delete()` I don't believe this is necessary to have an `await` here.